coordinator/src/node.rs

```diff
@@ -201,6 +192,34 @@ enum TradeAction {
     // Reduce,
 }
 
+/// Calculates the accept settlement amount based on the pnl.
+///
+/// The smallest possible result is 0.
+fn calculate_accept_settlement_amount(
+    opening_price: Decimal,
+    closing_price: Decimal,
+    quantity: f64,
+    long_leverage: f64,
+    short_leverage: f64,
+    direction: Direction,
+) -> Result<u64> {
+    let pnl = cfd::calculate_pnl(
+        opening_price,
+        closing_price,
+        quantity,
+        long_leverage,
+        short_leverage,
+        direction,
+    )?;
+
+    let margin_trader = calculate_margin(opening_price, quantity, long_leverage);
```

Comment:
🛠️ I think it is incorrect to assume that the `margin_trader` is always using the `long_leverage`, we would have to determine the leverage to be used based on the direction I suppose.

Reply:
indeed 🙈 good find 👍